src/psyclone/parse/module_info.py

```diff
@@ -255,33 +267,33 @@ def get_psyir(self):
             exist in the PSyIR.
 
         '''
-        if self._psyir is None:
+        if self._psyir_container_node is None:
+            # Keep this try-except for legacy reasons
             try:
-                ptree = self.get_parse_tree()
-            except FortranSyntaxError as err:
+                self._file_info.get_fparser_tree()
+            except FileInfoFParserError as err:
                 # TODO #11: Add proper logging
                 print(f"Error parsing '{self.filename}': '{err}'")
                 return None
-            if not ptree:
-                # TODO #11: Add proper logging
-                print(f"Empty parse tree returned for '{self.filename}'")
-                return None
+
             try:
-                self._psyir = self._processor.generate_psyir(ptree)
-            except (KeyError, SymbolError, InternalError, GenerationError) \
-                    as err:
+                self._psyir_container_node = self._file_info.get_psyir_node()
```

Comment:
I'd like to keep it split because depending on whether the fparser or psyir failed, the logging information would be different.
Anyhow, I replaced the except (..., ..., ...) blocks with just `except FileInfoFParserError`.
The raised exceptions are now also clearer (build up step-by-step by appending the previous exception message)